src/common/reader.js

```diff
@@ -186,6 +186,10 @@ class Reader {
 				entireWord: false,
 				result: null
 			},
+			a11yVirtualCursorTarget: {
```

Comment:
Why do we need to expose nodes from within the view’s iframe? Shouldn’t the view itself track and manage its virtual cursor independently from the reader UI? It seems like all the `a11yVirtualCursorTarget` related code could be moved into the views.

What happens if the view is split and both views modify the `a11yVirtualCursorTarget`?

Reply:
That actually is a fair point... I'll offload the rest of virtual cursor logic into the views.

I thought of the split views as a very unlikely edge case because this is done just to help screen readers figure out where to look, so I think a split view is somewhat unlikely in this context. In a scenario that it does happen, I thought the later update would get the priority (as it's the view currently worked on), and if through some obscure focus movements we enter a view that the cursor target does not belongs to, we do [nothing](https://github.com/zotero/reader/blob/6d411f174335e887ec58c46f2a7ee021720e15fe/src/common/reader.js#L1002). But, yeah, if each frame keeps track of the node itself, this is no longer relevant.

Reply:
But after removing `a11yVirtualCursorTarget` it now should be possible to move `placeA11yVirtualCursor` into views as well? Because it simply retrieves a target element from the view and doesn’t appear to interact with anything outside the iframe. If it needs to be triggered on focus, note that the views have a `focus()` method.

Reply:
It totally is. `placeA11yVirtualCursor` is the same for all views though (at least as of now) and has some amount of logic. So I thought it would be better to avoid duplicating code in case we want to change something later to avoid having to edit it in multiple places. There is some duplication now already for the getter and setter of the `a11yVirtualCursorTarget` but those don't have as much logic so I figured those are not a big issue.

Is duplication OK in this instance? Or is there a better place for views to share this function?

Reply:
If you just want to reuse the same function across views, you can add it to `src/common/lib/utilities.js`.

Reply:
Ok, gotcha. I moved `placeA11yVirtualCursor` to utils, so now we don't have anything `virtualCursorTarget`-related in the Reader. With some additional refactoring to simplify things, all the logic is pushed into the views, and there is actually pretty much no duplication left. So that's good!